src/compile/repeater.ts

```diff
@@ -44,16 +63,26 @@ function replaceRepeater(mapping: EncodingOrFacet<Field>, repeater: RepeaterValu
   const out: EncodingOrFacet<string> = {};
   for (const channel in mapping) {
     if (mapping.hasOwnProperty(channel)) {
-      const fieldDef: FieldDef<Field> | FieldDef<Field>[] = mapping[channel];
+      const channelDef: ChannelDef<Field> = mapping[channel];
```

Comment:
Hmm, `detail` and `order` can be an array of fieldDef too.

Reply:
Can they be conditionals?